pkg/operator/operator.go

```diff
@@ -749,7 +752,8 @@ func (o *Operator) sync(ctx context.Context, key string) error {
 				newTaskSpec("NodeExporter", tasks.NewNodeExporterTask(o.client, factory)),
 				newTaskSpec("KubeStateMetrics", tasks.NewKubeStateMetricsTask(o.client, factory)),
 				newTaskSpec("OpenshiftStateMetrics", tasks.NewOpenShiftStateMetricsTask(o.client, factory)),
-				newTaskSpec("PrometheusAdapter", tasks.NewPrometheusAdapterTask(ctx, o.namespace, o.client, factory, config)),
+				newTaskSpec("PrometheusAdapter", tasks.NewPrometheusAdapterTask(ctx, o.namespace, o.client, o.prometheusAdapterEnabled, factory, config)),
```

Comment:
(nit) I'd negate o.metricsServerEnabled
```suggestion
				newTaskSpec("PrometheusAdapter", tasks.NewPrometheusAdapterTask(ctx, o.namespace, o.client, !o.metricsServerEnabled, factory, config)),
```

Reply:
I did negate but moved to a new var https://github.com/openshift/cluster-monitoring-operator/pull/2022/files#diff-99c0290799daf9abc6240df64063e20bfaf67b371577b67ac7eec6f4725622ffR439, we don't need that?

Reply:
discussed offline, decided to keep only `o.metricsServerEnabled` field